modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathLocator.cs

```diff
@@ -1,185 +1,293 @@
+
+    internal interface IRiderLocatorEnvironment
+    {
+        OperatingSystemFamily operatingSystemFamily { get; }
+
+        T FromJson<T>(string json);
+        void Warn(string message, Exception e = null);
+        void LogException(Exception exception);
+    }
+
     /// <summary>
     /// This code is a modified version of the JetBrains resharper-unity plugin listed under Apache License 2.0 license:
     /// https://github.com/JetBrains/resharper-unity/blob/master/unity/JetBrains.Rider.Unity.Editor/EditorPlugin/RiderPathLocator.cs
     /// </summary>
-    public static class RiderPathLocator
+    internal class RiderPathLocator
     {
-        public static RiderInfo[] GetAllRiderPaths()
+        public readonly IRiderLocatorEnvironment RiderLocatorEnvironment;
+
+        internal RiderPathLocator(IRiderLocatorEnvironment riderLocatorEnvironment)
+        {
+            RiderLocatorEnvironment = riderLocatorEnvironment;
+        }
+
+        [UsedImplicitly] // Used in com.unity.ide.rider
+        public RiderInfo[] GetAllRiderPaths()
         {
             try
             {
-                if (OS.IsWindows)
-                {
-                    return CollectRiderInfosWindows();
-                }
-                if (OS.IsMacOS)
+                switch (RiderLocatorEnvironment.operatingSystemFamily)
                 {
-                    return CollectRiderInfosMac();
+                    case OperatingSystemFamily.Windows:
+                        return CollectRiderInfosWindows();
+                    case OperatingSystemFamily.MacOSX:
+                        return CollectRiderInfosMac();
+                    case OperatingSystemFamily.Linux:
+                        return CollectAllRiderPathsLinux();
                 }
-                if (OS.IsUnixLike)
-                {
-                    return CollectAllRiderPathsLinux();
-                }
-                throw new InvalidOperationException("Unexpected OS.");
             }
             catch (Exception e)
             {
-                GD.PushWarning(e.Message);
+                RiderLocatorEnvironment.LogException(e);
             }
 
-            return Array.Empty<RiderInfo>();
+            return new RiderInfo[0];
         }
 
-        private static RiderInfo[] CollectAllRiderPathsLinux()
+        private RiderInfo[] CollectAllRiderPathsLinux()
         {
             var installInfos = new List<RiderInfo>();
-            string home = Environment.GetEnvironmentVariable("HOME");
+            var appsPath = GetAppsRootPathInToolbox();
+
+            installInfos.AddRange(CollectToolbox20Linux(appsPath, "*rider*", "bin/rider.sh"));
+            installInfos.AddRange(CollectToolbox20Linux(appsPath, "*fleet*", "bin/Fleet"));
+
+            var riderRootPath = Path.Combine(appsPath, "Rider");
+            installInfos.AddRange(CollectPathsFromToolbox(riderRootPath, "bin", "rider.sh", false)
+                .Select(a => new RiderInfo(this, a, true)).ToList());
```

Comment:
You shouldn't need `.ToList` here, the `List<T>.AddRange` method takes an `IEnumerable<T>` parameter.

```suggestion
                .Select(a => new RiderInfo(this, a, true)));
```

Same thing applies to all other instances of this throughout this file.